### PR TITLE
Feature/visualization

### PR DESCRIPTION
**Pull request by mengweieric**
### Description
Added visualization related features.

### Issues Resolved
- Supports rendering visualization based on PPL query [#78]
- Explorer visualization panel [#53]

### Additional Notes
- Having intermittent issues with legacy opensearch client when processing concurrent requests, will fix it later.
- Query/visualization history is dummy now for demo purpose

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by joshuali925 — APPROVED** [review]

File: public/common/types/explorer.ts [hunk 1]
```diff
@@ -22,9 +22,9 @@ export interface IQueryTab {
   content: React.ReactNode
 }
 
-export interface ILogExplorerProps {
-  http: CoreStart['http']
-}
+// export interface ILogExplorerProps {
```
Comment (davidcui1225) [inline public/common/types/explorer.ts:25]: Safe to remove the commented export here?

Comment (mengweieric) [reply public/common/types/explorer.ts:25]: yes, I'll remove.

File: public/components/explorer/explorer.tsx [hunk 2]
```diff
@@ -94,6 +113,20 @@ export const Explorer = ({
     [setFixedScrollEl]
   );
 
+  const getIndexFromRawQuery = (rawQuery: string) => rawQuery.split('=')[1].split(' ')[0];
+
+  useEffect(() => {
+    if (!query) return;
+    if (query.includes('stats')) {
```
[the remark this answers is not in the document]
Comment (davidcui1225) [reply public/components/explorer/explorer.tsx:120]: +1, had a similar check in dashboards-reports that ended up causing an issue and we had to change it to a stricter check.

File: public/components/explorer/visualizations/shared_components/toolbar_popover.tsx [hunk 3]
```diff
@@ -50,7 +50,7 @@ export const ToolbarPopover: React.FunctionComponent<ToolbarPopoverProps> = ({
   const [open, setOpen] = useState(false);
 
   const iconType: string | IconType = typeof type === 'string' ? typeToIconMap[type] : type;
-
+  console.log('iconType: ', iconType);
```
Comment (davidcui1225) [inline public/components/explorer/visualizations/shared_components/toolbar_popover.tsx:53]: np: remove console.log statement

File: server/adaptors/pplDatasource.ts [hunk 4]
```diff
@@ -43,21 +42,11 @@ export class PPLDataSource {
...
-          ) {
-            hasTimestamp = true;
-        }
       }
 
       data.push(record);
     });
-    pplRes['jsonData'] = data;
-    pplRes['hasTimestamp'] = hasTimestamp;
+    this.pplDataSource['jsonData'] = data;
```
Comment (davidcui1225) [inline server/adaptors/pplDatasource.ts:49]: What are the parameters of the `pplDataSource` object, how come `jsonData` is being called with bracket notation vs. dot notation? Should we add a Type or Interface for it?

File: public/components/explorer/explorer.tsx [hunk 5]
```diff
@@ -94,6 +113,20 @@ export const Explorer = ({
     [setFixedScrollEl]
   );
 
+  const getIndexFromRawQuery = (rawQuery: string) => rawQuery.split('=')[1].split(' ')[0];
```
[the remark this answers is not in the document]
Comment (mengweieric) [reply public/components/explorer/explorer.tsx:116]: Yes, this makes sense. We have plan to make autocomplete add index pattern to state data whenever it is selected through autosuggestion, and access the index directly without processing raw query string. I think once that part is finished, then here we don't need to split the string in order to get the index. But still we'll need to cover corner cases where user just paste a query without selecting index, then it needs to parse the raw query. I'll use regex.

File: public/components/explorer/explorer.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (mengweieric) [reply public/components/explorer/explorer.tsx:120]: Yes, I'll use regex.

File: public/components/explorer/visualizations/shared_components/toolbar_popover.tsx (hunk 3 above)
Comment (mengweieric) [reply public/components/explorer/visualizations/shared_components/toolbar_popover.tsx:53]: Right, thanks, I'll remove.

File: server/adaptors/pplDatasource.ts (hunk 4 above)
Comment (mengweieric) [reply server/adaptors/pplDatasource.ts:49]: The pplDataSource is just the response object returned from opensearch with schema-to-value mapping added. I'll add interface for it.

**Pull request merged**